node/v2/handler.js

```diff
@@ -203,6 +211,13 @@ TChannelV2Handler.prototype.handleCallResponse = function handleCallResponse(res
         return callback(new Error('call response before init response')); // TODO typed error
     }
     var res = self.buildInResponse(resFrame);
+    if (resFrame.body.args && resFrame.body.args[0] &&
+        resFrame.body.args[0].length > v2.CallResponse.MaxArg1Size) {
+        return callback(errors.Arg1OverLengthLimit({
+                length: '0x' + resFrame.body.args[0].length.toString(16),
+                limit: '0x' + v2.CallResponse.MaxArg1Size.toString(16)
```

Comment:
@jcorbin mentioned that calling this callback with an error constitutes a protocol error which would cause this connection to be torn down. This seems appropriate to me since the stream is not longer in a consistent state.

On a side note, if we go with UDP in v2+, connection scoped identifiers would become irrelevant, protocol errors would not break the connection, and we would need to use tracing to identify a req/res pair.